clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+
+All of these events are surfaced by the DirectLineSpeechDialogBackend and handled in DialogManager. For activities, DialogManager uses a queue, the [DialogResponseQueue](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DialogResponseQueue.cs), to make sure activities are executed sequentially. For example, if a bot responded with two activities, the second activity's text and audio would be queued while the audio from the first activity is played then dequeued when the first activity finishes.
+
+### Subsequent Turns
+If a bot requests a response to the last turn (like if it asked a question and expects the user to answer) then the DialogResponse.FollowupTurnIndicated will be set to true. When this field is true, the app just needs to start listening for user input again as if it received an activation. 
+
+DialogManager saves the value of the DialogResponse.FollowupTurnIndicated field during its OnActivityReceived event handler. When the DialogResponseQueue indicates that the last turn has finished executing, DialogManager calls FinishTurnAsync. If FollowupTurnIndicated is true, then it initiates listening with StartTurnAsync and starts providing audio to the DirectLineSpeechDialogBackend again.
+
+## Reporting Agent State
+Throughout activation, keyword verification, and the bot interaction, a UWP voice agent application needs to update MVA with its state. MVA uses this information to manage multiple voice agents running at the same time, system state and lighting, and other important functions. There are 4 states to report to MVA.
+- Inactive: The application is inactive
+- Detecting: Keyword verification is in progress
+- Listening: The voice agent is listening to the user
```

Comment:
Also:

- `**Working**`: The agent is taking action based on what the user said